main.tf

```
@@ -82,9 +81,8 @@ resource "azurerm_storage_account" "this" {
         }
       }
       dynamic "delete_retention_policy" {
-        for_each = blob_properties.value.delete_retention_policy == null ? [] : [
-          blob_properties.value.delete_retention_policy
-        ]
+        for_each = (
+        blob_properties.value.delete_retention_policy != null && try(blob_properties.value.delete_retention_policy.enabled, true)) ? [blob_properties.value.delete_retention_policy] : []
```

If I am not mistaken, it is impossible for this to be null.

```suggestion
        blob_properties.value.delete_retention_policy.enabled ? [blob_properties.value.delete_retention_policy] : []
```